pkg/appgw/certificates.go

```diff
@@ -115,7 +115,7 @@ func (c *appGwConfigBuilder) newHostToSecretMap(ingress *v1beta1.Ingress) map[st
 		}
 
 		// default secret
-		if len(tls.Hosts) == 0 {
+		if len(tls.Hosts) == 0 || len(ingress.Spec.Rules) == 0 {
```

Comment:
This should be independent of ingress.Rules. I think this check will break ingress with rules without host.

Reply:
> This should be independent of ingress.Rules. I think this check will break ingress with rules without host.

this check only happens when TLS defined, no matter you have ingress hosts or not

Reply:
yes, we can treat this as loose binding. We can treat them independently.